src/libraries/Microsoft.PowerFx.Core/Syntax/Formula.cs

```diff
@@ -40,6 +40,17 @@ public Formula(string script, ILanguageSettings loc = null)
             AssertValid();
         }
 
+        public Formula(string script, TexlNode tree, ILanguageSettings loc = null)
```

Comment:
nit: We could use constructor chaining to share the impl between ctors:

 public Formula(string script, ILanguageSettings loc = null) 
 : this(script, null, loc) 
{ }